src/main/kotlin/racingcar/domain/Race.kt

```diff
@@ -0,0 +1,29 @@
+package racingcar.domain
+
+class Race(private val raceSetting: RaceSetting) {
+
+    private val cars: List<Car> = List(raceSetting.numberOfCar) { index ->
+        Car("${index + 1}번째 자동차")
+    }
+
+    private val roundHistories = mutableListOf<RoundHistory>()
+
+    fun run(): RaceResult {
+        repeat(raceSetting.totalCountOfRound) { roundNumber: Int ->
+            startRound(cars)
+            saveRoundHistory(roundNumber, cars)
+        }
+
+        return RaceResult(roundHistories)
+    }
+
+    private fun startRound(cars: List<Car>) {
+        Round(cars).start()
+    }
+
+    private fun saveRoundHistory(roundNumber: Int, cars: List<Car>) {
+        cars.forEach { car ->
+            roundHistories.add(RoundHistory(roundNumber, car.name, car.position))
+        }
```

Comment:
아래와 같이 변경해볼 수도 있을 것 같아요. 😃😉
```suggestion
        val currentRoundHistories = cars.map { RoundHistory(roundNumber, it.name, it.position) }
        roundHistories.addAll(currentRoundHistories)
```

Reply:
개인적으로 RoundHistory 객체가 각각의 Car 객체 History를 찍는게 아니라 Round의 모든 Car 객체의 정보들을 가지고 있으면 어떨까요? 🤔 

예를들어 RoundHistory가 아래와 같이 `roundNumber`와 각 `carInfos` (명칭은 그냥 떠오른대로 언급했습니다 😅 참고만 해주세요)를 가지도록 해볼 수 있을 것 같아요. 😃

```kotlin
data class RoundHistory(
    val round: Int,
    val carInfos: List<Pair<String, Int>>
)
```